### DISCUSSION TITLE
When sampling mechanism is off, why the method trySampling always return true ?

### DISCUSSION THREAD

**Discussion by thisisgpy** [apache/skywalking#8217] [Q&A]
When sampling mechanism is off, why the method `org.apache.skywalking.apm.agent.core.sampling.SamplingService#trySampling`  always return `true` ?

```
/**
 * @param operationName The first operation name of the new tracing context.
 * @return true, if sampling mechanism is on, and getDefault the sampling factor successfully.
 */
public boolean trySampling(String operationName) {
    if (on) {
        int factor = samplingFactorHolder.get();
        if (factor < samplingRateWatcher.getSamplingRate()) {
            return samplingFactorHolder.compareAndSet(factor, factor + 1);
        } else {
            return false;
        }
    }
    return true;
}
```

**Comment by wu-sheng** [comment]
What do you mean? What do you expect?

**Reply by thisisgpy** [reply]
> What do you mean? What do you expect?

From the method's comments, this method will return true only when the sampling mechanism is enabled and the `samplingFactorHolder` is successfully increased. But in the code, it will return true when the sampling mechanism is disabled. So what does the return value of this method mean? I think that only when true is returned, it means that a link can be sampled. But why is sampling allowed even when the sampling mechanism is enable?

**Reply by wu-sheng** [reply]
This comment needs some adjustments. It was 4 years ago, that comment is correct when sampling mechanism is ON. 
The short version is `return true` means `should sample`.

Then correct comment is, welcome you to submit a pull request to correct this comment.

```
    /**
     * @param operationName The first operation name of the new tracing context.
     * @return true if should sample this trace segment. When sampling mechanism is on, return true if sample limited is not reached.
     */
```

**Reply by thisisgpy** [reply]
> This comment needs some adjustments. It was 4 years ago, that comment is correct when sampling mechanism is ON. The short version is `return true` means `should sample`.
> 
> Then correct comment is, welcome you to submit a pull request to correct this comment.
> 
> ```
>     /**
>      * @param operationName The first operation name of the new tracing context.
>      * @return true if should sample this trace segment. When sampling mechanism is on, return true if sample limited is not reached.
>      */
> ```

What I wonder is why it returns true when the sampling mechanism is disabled (when `on = false`).

**Reply by wu-sheng** [reply]
Sampling is OFF mean sampling 100% agent. If you don't want to sample any trace, uninstall the agent.

**Reply by wu-sheng** [reply]
Follow up https://github.com/apache/skywalking/issues/8227
